src/main.js

```diff
@@ -38,22 +36,25 @@ function saveIdea(event) {
   }
 
   newIdea = new Idea(title.value, body.value, false, Date.now());
-  
-  newIdea.saveToStorage(newIdea);
+
+  newIdea.saveToStorage();
   ideas.push(newIdea);
   render(ideas);
   clearInputs();
   saveButton.classList.add('disable-save');
 }
 
 function renderPage() {
-  var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));
-  
-  for (var i = 0; i < retrievedIdeas.length; i++) {
-    newIdea = new Idea(retrievedIdeas[i].title, retrievedIdeas[i].body, retrievedIdeas[i].star, retrievedIdeas[i].id);
-    ideas.push(newIdea);
+  //handles error on load for null array
+  var storedIdeas = JSON.parse(localStorage.getItem("ideas"));
+  if (storedIdeas) {
+    for (var i = 0; i < storedIdeas.length; i++) {
+      //returns array as an array of instances of the idea class
+      newIdea = new Idea(storedIdeas[i].title, storedIdeas[i].body, storedIdeas[i].star, storedIdeas[i].id);
+      ideas.push(newIdea);
+    }
   }
-  
+  localStorage.setItem("ideas", JSON.stringify(ideas));
```

Comment:
This line isn't needed - we only call renderPage once and it's pulling ideas from storage, it doesn't remove them from storage so we don't need to reset them.

Reply:
Good call--just deleted it